internal/lotus/discovery.go

```diff
@@ -4,86 +4,123 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"net/url"
 
 	"github.com/filecoin-project/go-address"
-	"github.com/filecoin-project/lotus/api"
-	"github.com/filecoin-project/lotus/chain/actors/builtin/miner"
 	"github.com/filecoin-project/storetheindex/internal/providers/discovery"
+	"github.com/ipfs/go-cid"
 	"github.com/libp2p/go-libp2p-core/peer"
 	"github.com/multiformats/go-multiaddr"
-	// If it is necessary to have lotus verify a signature
-	//"github.com/filecoin-project/go-state-types/crypto"
-	//"github.com/filecoin-project/lotus/tree/master/lib/sigs"
+	jrpc "github.com/ybbus/jsonrpc/v2"
 )
 
-type Discovery struct {
-	node   api.Gateway
-	closer *nodeCloser
+type Discoverer struct {
+	gatewayURL string
 }
 
-func (d *Discovery) Close() error {
-	return d.closer.Close()
+type ExpTipSet struct {
+	Cids []cid.Cid
+	//Blocks []*BlockHeader
+	//Height abi.ChainEpoch
+	Blocks []interface{}
+	Height int64
 }
 
-func (d *Discovery) Discover(ctx context.Context, peerID peer.ID, minerAddr string, data, signature []byte) (*discovery.Discovered, error) {
-	tipSet, err := d.node.ChainHead(ctx)
+type MinerInfo struct {
+	Owner                      address.Address
+	Worker                     address.Address
+	NewWorker                  address.Address
+	ControlAddresses           []address.Address
+	WorkerChangeEpoch          int64
+	PeerId                     *peer.ID
+	Multiaddrs                 [][]byte
+	WindowPoStProofType        int64
+	SectorSize                 uint64
+	WindowPoStPartitionSectors uint64
+	ConsensusFaultElapsed      int64
+}
+
+// New creates a new lotus Discoverer
+func NewDiscoverer(gateway string) (*Discoverer, error) {
+	u, err := url.Parse(gateway)
 	if err != nil {
-		return nil, fmt.Errorf("cannot get filecoin chain head: %s", err)
+		return nil, err
 	}
-	tipSetKey := tipSet.Key()
+	u.Scheme = "https"
+	u.Path = "/rpc/v1"
 
+	return &Discoverer{
+		gatewayURL: u.String(),
+	}, nil
+}
+
+func (d *Discoverer) Discover(ctx context.Context, peerID peer.ID, minerAddr string) (*discovery.Discovered, error) {
```

Comment:
Why are we using the peer.ID as an argument? Aren't we getting this info from lotus? (I don't have a deep knowledge of Lotus, just want to understand how it works :) )